OpenRA.Mods.Common/Effects/SpriteEffect.cs

```diff
@@ -20,25 +21,44 @@ public class SpriteEffect : IEffect, ISpatiallyPartitionable
 		readonly World world;
 		readonly string palette;
 		readonly Animation anim;
-		readonly WPos pos;
+		readonly Func<WPos> posFunc;
 		readonly bool visibleThroughFog;
 		readonly bool scaleSizeWithZoom;
+		readonly bool staticPosition;
+		WPos pos;
 
+		// Facing is last on these overloads partially for backwards compatibility with previous main ctor revision
+		// and partially because most effects don't need it.
 		public SpriteEffect(WPos pos, World world, string image, string sequence, string palette, bool visibleThroughFog = false, bool scaleSizeWithZoom = false, int facing = 0)
+			: this(() => pos, () => facing, world, image, sequence, palette, visibleThroughFog, scaleSizeWithZoom)
+		{
+			staticPosition = true;
```

Comment:
Evaluating `posFunc` has a negligible perf impact, so we could remove this and keep things simpler.